File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/webapp/src/ds/ImportMapItemDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.ad.impex.ds.model.ImportMapItemDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"importMapId", type:"int", useNull:true},
		{name:"importMapName", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"dataSource", type:"string"},
		{name:"path", type:"string"},
		{name:"fileName", type:"string"},
		{name:"ukFieldName", type:"string"},
		{name:"active", type:"boolean"}
	]
});

Ext.define("net.nan21.dnet.module.ad.impex.ds.model.ImportMapItemDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"importMapId", type:"int", useNull:true},
		{name:"importMapId_From",type:"int", useNull:true},
		{name:"importMapId_To",type:"int", useNull:true},
		{name:"importMapName", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"sequenceNo_From",type:"int", useNull:true},
		{name:"sequenceNo_To",type:"int", useNull:true},
		{name:"dataSource", type:"string"},
		{name:"path", type:"string"},
		{name:"fileName", type:"string"},
		{name:"ukFieldName", type:"string"},
		{name:"active", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.ad.impex.ds.param.ImportMapItemDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"importFileLocation", type:"string"},
		{name:"importDataSource", type:"string"},
		{name:"importUkFieldName", type:"string"}
	]
}); 
 

File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/impex/ds/model/ImportMapItemDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.ad.impex.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;

import net.nan21.dnet.module.ad.impex.domain.entity.ImportMapItem;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = ImportMapItem.class, sort = { @SortField(field = ImportMapItemDs.fSEQUENCENO) })
public class ImportMapItemDs extends AbstractAuditableDs<ImportMapItem> {

    public static final String fIMPORTMAPID = "importMapId";
    public static final String fIMPORTMAPNAME = "importMapName";
    public static final String fSEQUENCENO = "sequenceNo";
    public static final String fDATASOURCE = "dataSource";
    public static final String fFILENAME = "fileName";
    public static final String fUKFIELDNAME = "ukFieldName";
    public static final String fACTIVE = "active";

    @DsField(join = "left", path = "importMap.id")
    private Long importMapId;

    @DsField(join = "left", path = "importMap.name")
    private String importMapName;

    @DsField()
    private Integer sequenceNo;

    @DsField(path = "dataSourceName")
    private String dataSource;

    @DsField()
    private String fileName;

    @DsField()
    private String ukFieldName;

    @DsField()
    private Boolean active;

    public ImportMapItemDs() {
        super();
    }

    public ImportMapItemDs(ImportMapItem e) {
        super(e);
    }

    public Long getImportMapId() {
        return this.importMapId;
    }

    public void setImportMapId(Long importMapId) {
        this.importMapId = importMapId;
    }

    public String getImportMapName() {
        return this.importMapName;
    }

    public void setImportMapName(String importMapName) {
        this.importMapName = importMapName;
    }

    public Integer getSequenceNo() {
        return this.sequenceNo;
    }

    public void setSequenceNo(Integer sequenceNo) {
        this.sequenceNo = sequenceNo;
    }

    public String getDataSource() {
        return this.dataSource;
    }

    public void setDataSource(String dataSource) {
        this.dataSource = dataSource;
    }

    public String getFileName() {
        return this.fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getUkFieldName() {
        return this.ukFieldName;
    }

    public void setUkFieldName(String ukFieldName) {
        this.ukFieldName = ukFieldName;
    }

    public Boolean getActive() {
        return this.active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

}
